gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -4307,6 +4325,25 @@ gnc_plugin_page_register_cmd_execassociated_transaction (GtkAction *action,
 
 }
 
+static void
+gnc_plugin_page_register_cmd_execassociated_invoice (GtkAction *action,
+        GncPluginPageRegister *plugin_page)
+{
+    GncPluginPageRegisterPrivate *priv = GNC_PLUGIN_PAGE_REGISTER_GET_PRIVATE(plugin_page);
+    SplitRegister *reg = gnc_ledger_display_get_split_register (priv->gsr->ledger);
+    Split *split = gnc_split_register_get_current_split (reg);
+    Transaction *trans = xaccSplitGetParent(split);
+    GncInvoice *invoice = gncInvoiceGetInvoiceFromTxn(trans);
+
+    ENTER("(action %p, plugin_page %p)", action, plugin_page);
+
+    g_return_if_fail(GNC_IS_PLUGIN_PAGE_REGISTER(plugin_page));
```

Comment:
You are already using plugin_page for GncPluginPageRegisterPrivate so the g_return_if_fail is not doing its job. See the blank transaction function for example.

Reply:
so, remove this line?

Reply:
No, do this...
    GncPluginPageRegisterPrivate *priv;
    SplitRegister *reg;
    Split *split;
    Transaction *trans;
    GncInvoice *invoice;

    ENTER("(action %p, plugin_page %p)", action, plugin_page);

    g_return_if_fail(GNC_IS_PLUGIN_PAGE_REGISTER(plugin_page));

   priv = GNC_PLUGIN_PAGE_REGISTER_GET_PRIVATE(plugin_page);
   reg = gnc_ledger_display_get_split_register (priv->gsr->ledger);

etc...

and you could just do this if you wanted...
    invoice = gncInvoiceGetInvoiceFromTxn (xaccSplitGetParent (gnc_split_register_get_current_split (reg)));

and get rid of 'split' and 'trans' unless you want to test for a valid 'split' and 'trans'